bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/internal/loader/ScriptFileWatcher.java

```diff
@@ -252,9 +253,17 @@ private void checkFiles() {
         SortedSet<URL> reimportUrls = new TreeSet<URL>(new Comparator<URL>() {
             @Override
             public int compare(URL o1, URL o2) {
-                String f1 = o1.getPath();
-                String f2 = o2.getPath();
-                return String.CASE_INSENSITIVE_ORDER.compare(f1, f2);
+                Path path1 = Paths.get(o1.getPath());
+                Path path2 = Paths.get(o2.getPath());
+                String name1 = path1.getFileName().toString();
+                String name2 = path2.getFileName().toString();
+                int nameCompare = name1.compareToIgnoreCase(name2);
+                if (nameCompare != 0) {
+                    return nameCompare;
+                } else {
+                    int pathCompare = path1.getParent().toString().compareToIgnoreCase(path2.getParent().toString());
+                    return pathCompare;
```

Comment:
You should not create a temporary local variable if you only intend to return it immediately.
But lets move forward and fix it another time.

Reply:
OK, thank you. I thought I had seen similar done elsewhere in the code. I will keep this in mind in the future.